src/hooks/useRankingData.js

```diff
@@ -54,14 +71,44 @@ const useRankingData = () => {
     })();
   }, [today]);
 
+  useEffect(() => {
+    (async function () {
+      const q = query(collection(firebaseDb, USERS_STATISTICS), orderBy('currentStreak', 'desc'));
+      const docs = await getDocs(q);
+      const results = [];
+      let position = 0;
+      let currentStreakValue = Number.MAX_SAFE_INTEGER;
+      docs.forEach(async doc => {
+        const { id: userEmail } = doc;
+        const { currentStreak, lastDatePlayed, totalGames } = doc.data();
+        if (!!totalGames) {
+          if (currentStreak !== currentStreakValue) {
+            currentStreakValue = currentStreak;
+            position += 1;
+          }
+          results.push({
+            currentStreak,
+            lastDatePlayed,
+            position,
+            user: users[userEmail] || { email: userEmail, name: userEmail.split('@')[0] },
+          });
+        }
+      });
+      setCurrentStreakRankingData(results);
+      setLoading(false);
+    })();
+  }, [users]);
+
   const currentUserPlayed = dailyResults.find(item => item.user.email === currentUser);
 
   return {
     currentUser,
     currentUserPlayed,
+    currentStreakRankingData,
     dailyResults,
     expandedUser,
     setExpandedUser,
```

Comment:
Hey, it is a bad practice to allow components to directly handle the state of others. The best option here it to define a function that receives as an argument the new value to update the state. By doing that we can control the state and its updates.

Reply:
After I change the way expanded works, I will make sure this doesn't happen anymore 😉